### PR TITLE
Adding power saving mode for scaling in sample decode.

### PR DESCRIPTION
**Pull request by FurongZhang**
Signed-off-by: Furong Zhang <furong.zhang@intel.com>

### CODE REVIEW

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 1]
```diff
@@ -573,6 +573,35 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
...
+            mfxExtVPPDoUse extDoUse;
+            memset(&extDoUse, 0, sizeof(extDoUse));
+            mfxU32 tabDoUseAlg[1];
+            extDoUse.Header.BufferId = MFX_EXTBUFF_VPP_DOUSE;
+            extDoUse.Header.BufferSz = sizeof(mfxExtVPPDoUse);
+            extDoUse.NumAlg = 1;
+            extDoUse.AlgList = tabDoUseAlg;
+            tabDoUseAlg[0] = MFX_EXTBUFF_VPP_DENOISE;
+
+            mfxExtVPPScaling scalingConfig;
```
Comment (AntonGrishin) [inline samples/sample_decode/src/pipeline_decode.cpp:592]: Please use m_mfxVppVideoParams.AddExtBuffer like here:
https://github.com/Intel-Media-SDK/MediaSDK/blob/55381eec2dfc47ec8190f7fe4e52ee18991cd3ce/samples/sample_decode/src/pipeline_decode.cpp#L972

File: samples/sample_decode/src/sample_decode.cpp [hunk 2]
```diff
@@ -272,6 +272,10 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
         {
             pParams->bPerfMode = true;
         }
+        else if (0 == msdk_strcmp(strInput[i], MSDK_STRING("-powersaving")))
```
Comment (AntonGrishin) [inline samples/sample_decode/src/sample_decode.cpp:275]: Please add this key to sample help in this file, and to the documentation https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/samples/readme-decode_linux.md

**Review by AntonGrishin — CHANGES_REQUESTED** [review]

File: samples/sample_decode/src/pipeline_decode.cpp (hunk 1 above)
Comment (dmitryermilov) [reply samples/sample_decode/src/pipeline_decode.cpp:592]: Denoise? why?

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 3]
```diff
@@ -573,6 +573,35 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
...
+            memset(&scalingConfig, 0, sizeof(scalingConfig));
+            scalingConfig.Header.BufferId = MFX_EXTBUFF_VPP_SCALING;
+            scalingConfig.Header.BufferSz = sizeof(mfxExtVPPScaling);
+            scalingConfig.ScalingMode = MFX_SCALING_MODE_LOWPOWER;        // low power mode
```
Comment (dmitryermilov) [inline samples/sample_decode/src/pipeline_decode.cpp:596]: It's weird to see mfxExtVPPScaling for CSC case (this patch is motivated by internal thread where CSC is discussed). If mfxExtVPPScaling  can turns on VEBOX+SFC for CSC, it requires some clarification in the spec.

**Review by dmitryermilov — CHANGES_REQUESTED** [review]

File: samples/sample_decode/src/pipeline_decode.cpp (hunk 3 above)
Comment (FurongZhang) [reply samples/sample_decode/src/pipeline_decode.cpp]: Sorry, denoise is a typo. I changed that.

Comment (FurongZhang) [reply samples/sample_decode/src/pipeline_decode.cpp]: ok. Thanks, I will try to do some clarification in the spec today or early next week. Will go home now. Have a good day @dmitryermilov

Comment (dmitryermilov) [reply samples/sample_decode/src/pipeline_decode.cpp]: Thanks:) You too!

Comment (FurongZhang) [reply samples/sample_decode/src/pipeline_decode.cpp]: I've added some clarification in document and code. Please review.

File: samples/sample_decode/src/sample_decode.cpp (hunk 2 above)
Comment (dmitryermilov) [inline samples/sample_decode/src/sample_decode.cpp:275]: @FurongZhang , let's call the option scaling_mode::default, scaling_mode::lowpower, scaling_mode::quality to highlight that we're mapping this cmd option to mfxExtVPPScaling::ScalingMode

File: samples/sample_decode/src/pipeline_decode.cpp [hunk 4]
```diff
@@ -573,6 +573,30 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
...
+            vppExtParams->NumAlg = 1;
+            if (NULL == vppExtParams->AlgList)
+                vppExtParams->AlgList = new mfxU32 [vppExtParams->NumAlg];
+            if (!vppExtParams->AlgList)
+                return MFX_ERR_NULL_PTR;
+            vppExtParams->AlgList[0] = MFX_EXTBUFF_VPP_SCALING;
+
+            auto vppScalingParams = m_mfxVppVideoParams.AddExtBuffer<mfxExtVPPScaling>();
+            MSDK_CHECK_POINTER(vppScalingParams, MFX_ERR_MEMORY_ALLOC);
+            vppScalingParams->ScalingMode = MFX_SCALING_MODE_LOWPOWER;
```
Comment (dmitryermilov) [inline samples/sample_decode/src/pipeline_decode.cpp:597]: while this one is clear, can we avoid code between lines 582-593? Is it really needed to for ScalingMode?

File: samples/sample_decode/include/pipeline_decode.h [hunk 5]
```diff
@@ -115,6 +115,7 @@ struct sInputParams
     bool    outI420;
 
     bool    bPerfMode;
+    bool    bPowerSavingMode;
```
Comment (dmitryermilov) [inline samples/sample_decode/include/pipeline_decode.h:118]: Let's call it mfxU16 scalingMode;

**Comment by dmitryermilov** [comment]
@FurongZhang , could you address the comment?

**Comment by dmitryermilov** [comment]
Replaced with https://github.com/Intel-Media-SDK/MediaSDK/pull/2655

**Pull request closed without merging**